packages/base/src/UI5Element.js

```diff
@@ -793,13 +793,27 @@ class UI5Element extends HTMLElement {
 		return "";
 	}
 
+	/**
+	 * Hook that allows UI5 Web Components to prepare everything necessary before being defined
+	 * @protected
+	 * @returns {Promise<void>}
+	 */
+	static async onDefine() {
```

Comment:
hooks don't need to be defined on the base class. Also at the call site there is already a check if the hook is present, so better remove this one.